R/UnsupervisedTask.R

```diff
@@ -1,6 +1,6 @@
 makeUnsupervisedTask = function(type, data, weights = NULL, blocking = NULL, fixup.data = "warn", check.data = TRUE) {
   task = makeTask(type, data, weights, blocking, fixup.data = fixup.data, check.data = check.data)
-  checkTaskData(getTaskData(data))
+  checkTaskData(task$env$data)
```

Comment:
@jakob-r @berndbischl 
yes sorry I should have been more precise here.
````getTaskData(data)```` was simply a bug or a workaround of some sorts that worked before,
but did not do the trick anymore after I've done my changes to ````getTaskData()````.
I first tried ````checkTaskData(getTaskData(task))```` but then we run exactly into the problem
Bernd describes above, I guess. The traceback of the error is pretty tedious but as far is I understand we are then calling ````getTaskDescription(task)$target````, and the description object is not existing yet.
My next step was to look into ````makeSupervisedTask()````, where we also use ````task$env$data````, so I figured that's the right way to go here.
Thanks for looking through this with me!